src/ddspdrum/module.py

```diff
@@ -229,7 +229,7 @@ def __call__(self, mod_signal: np.array, phase: float = 0) -> np.array:
         cosine_argument = self.make_argument(control_as_frequency) + phase
 
         self.phase = cosine_argument[-1]
-        return np.cos(cosine_argument)
+        return self.oscillator(cosine_argument)
```

Comment:
The idea being, this can be inherited and overrided with difference oscillator types. Would it be cleaner to have a VCO base class and a SINE specific class to inherit? Whatever works...

Reply:
I'm into this